docs/NEW_LANGUAGE_FEATURE_INTEGRATION_PROCESS.md

```diff
@@ -69,25 +74,24 @@ A list to a few of the popular ones:
 
 **Examples**:
 
-- IntelliJ IDEA introduced an inspection rule for Java 17 or higher to detect
+- **Check updates**: IntelliJ IDEA introduced an inspection rule for Java 17 or higher to detect
   redundant `strictfp` modifiers. This resulted in the analysis of `RedundantModifierCheck`
   to verify if we needed to update it.
 
-### Real Usage Examples in Large Projects
+- **New check**: Sonar introduced a [new rule](https://rules.sonarsource.com/java/tag/java21/RSPEC-6916/) to detect the use of suggest use of `when` instead of
+  a single if statement inside a pattern match body.
+  This resulted in the creation of a new check `WhenShouldBeUsed` to enforce the
+  use of this new feature.
+
+## Real Usage Examples in Large Projects
 
 Review real usage examples of the new language feature in large projects.
 This helps to identify potential issues and ensures that the checks
 are aligned with practical use cases.
 A list of representative projects can be found
 [here](https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/github-action-projects1.properties).
 
-## New Check Procedure
-
-Creating new checks for new language features helps us establish
-best practices within the community, ensuring that developers
-utilize these features effectively.
-
-### Good Source of Best Practices
```

Comment:
I've been reflecting on the document and realized that separating check updates from new check creation might not be the most effective approach. Many methods are relevant to both updating existing checks and creating new ones, as evidenced by my recent tracker issues. Therefore, I propose combining these sections to streamline the process and provide examples for updating and new checks within each method (where possible).

Please share your feedback on this proposed change.

Reply:
It sure does seem like we have a fair amount of overlap, if you want to propose a different structure via a rough outline, I would like to see exactly what you have in mind. You can just do something simple right here, like this:

 - Some header
    - some section
    here we will talk about...
   - some other sections...
- Some other header...
   - blah blah...

Reply:
- Header (method name)
explanation of the method
Examples:
    - check updates example: blah blah   (where possible)
    - new checks example: blah blah  (where possible)

Reply:
@rnveach  hat do you think?

Reply:
https://github.com/checkstyle/checkstyle/pull/15465#discussion_r1715700977 is not clear to me what this is trying to convey. I don't understand how we went from new check and existing check documentation overlapping to a method name.

Reply:
> I don't understand how we went from new check and existing check documentation overlapping to a method name.

@rnveach I don't understand by method name I mean something like `Consider Similar and Related Tokens`. what I was trying to say instead of having separate  "check update" and "new checks" sections we should merge both. and for each method we let it talk for examples of check updates or new checks from its perspective

You guys can ignore my comments if it confuses you and take a look at the doc to understand my thoughts.

Reply:
I probably will understand better with a visual.

Reply:
From a high level, I think it makes sense to first consider how we can support new language features, then determine if we should create a new check or update an existing one.